### PR TITLE
Target .NET 6.0

### PR DESCRIPTION
**Pull request by bricelam**
Rationale:

* Unlike .NET Standard, older versions of .NET don't enable any additional platforms.
* We moved to .NET 5 because it enables the same platforms as .NET Standard 2.1, but with additional features (like savepoint APIs in ADO.NET) that we can leverage.
* .NET 5 will go out of support three months after .NET 6 ships anyway.
* The migration from .NET 5 to 6 should be pretty seamless

### CODE REVIEW

File: src/Microsoft.Data.Sqlite.Core/Microsoft.Data.Sqlite.Core.csproj [hunk 1]
```diff
@@ -15,7 +15,7 @@ Microsoft.Data.Sqlite.SqliteException
 Microsoft.Data.Sqlite.SqliteFactory
 Microsoft.Data.Sqlite.SqliteParameter
 Microsoft.Data.Sqlite.SqliteTransaction</Description>
-    <TargetFrameworks>netstandard2.0;net5.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
```
Comment (roji) [inline src/Microsoft.Data.Sqlite.Core/Microsoft.Data.Sqlite.Core.csproj:18]: If we want to, we could continue targeting net5.0 in addition to net6.0, this way users targeting net5.0 don't get jumped back to netstandard2.0 (and lose whatever newer things they were using). Or not.

**Review by roji — APPROVED** [review]

File: src/Microsoft.Data.Sqlite.Core/Microsoft.Data.Sqlite.Core.csproj (hunk 1 above)
Comment (bricelam) [reply src/Microsoft.Data.Sqlite.Core/Microsoft.Data.Sqlite.Core.csproj:18]: I don't think it's worth it. The API surface is the same between the two. A few APIs would require casting to the provider-specific types (won't work polymorphically) but they seem pretty corner case:

* SqliteBlob -- Calling via `Stream` is functionally the same, but copies more memory
  * Read(Span<byte>)
  * Write(ReadOnlySpan<byte>)
* SqliteTransaction -- Calling via `DbTransaction` will throw; `SupportsSavepoints` returns false
  * Save(string)
  * Rollback(string)
  * Release(string)

**Comment by ErikEJ** [comment]
What is the milestone for this?

**Comment by roji** [comment]
@ErikEJ this should apply for 6.0.0-preview6. Preview5 will still target net5.0.

**Comment by vslee** [comment]
One downside to doing this in the preview phase is that it makes it a little more difficult for people to test the previews. Before, ppl could just update the NuGet from 5.0 to a 6.0.0-preview and try out the new EF Core. Now, it also requires installing a preview .NET Core.

**Comment by roji** [comment]
@vslee that's true - we discussed this when making the decision to target net6.0. At the end of the day, we're starting to near the end of the 6.0 development cycle, and at some point we were bound to do this change. Hopefully installing a 6.0 dotnet SDK preview isn't that much of an effort.

**Pull request merged**